ads/inabox/position-observer.js

```diff
@@ -142,3 +144,27 @@ function getScrollingElement(win) {
 function isWebKit(ua) {
   return /WebKit/i.test(ua) && !/Edge/i.test(ua);
 }
+
+/**
+ * Get the element's layout rect relative to the viewport.
+ * If parentWin is provided, then attempt to walk up the DOM and add the
```

Comment:
shall we just always walk up to the host window?

Reply:
the parentWin parameter has to be provided, since this is a general-purpose position observer and there's no way to get the immediate window/iframe that contains the observed element.

I can change it so that if parentWin is not provided then automatically assume the host window to be its immediate parent. What do you think?

Reply:
Does `targetElement.ownerDocument.defaultView` not work?

Reply:
TIL. I will try that. If it works then the parameter wouldn't be needed.